### PR TITLE
feat(manifests): add 'app' label to workflow-controller-metrics service

### PR DESCRIPTION
**Pull request by dseynaev**
this PR adds an app label to the workflow-controller-metrics service for use in a ServiceMonitor resource when working with the prometheus operator:

https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/user-guides/getting-started.md#related-resources

it would probably be even better practice use `commonLabels` in the `kustomization.yaml` to set `app: workflow-controller` on all resources related to the workflow-controller

### CODE REVIEW

**Review by alexec — CHANGES_REQUESTED** [review]
I don't know why CI did not complain. Can you please run `make manifests` and commit changes?

**Comment by dseynaev** [comment]
@alexec I think there is a problem with one of the rules in the Makefile: `make manifests` did nothing until I fixed the dependency pattern

File: Makefile [hunk 1]
```diff
@@ -372,7 +372,7 @@ manifests/quick-start-mysql.yaml: /dev/null
 manifests/quick-start-postgres.yaml: /dev/null
 	dist/kustomize build --load_restrictor=none manifests/quick-start/postgres | ./hack/auto-gen-msg.sh > manifests/quick-start-postgres.yaml
 
-dist/manifests/%: manifests/$*
```
Comment (alexec) [inline Makefile:375]: 👀

**Review by alexec — APPROVED** [review]

**Comment by alexec** [comment]
LGTM

File: manifests/install.yaml [hunk 2]
```diff
@@ -197,16 +197,19 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: argo
+  namespace: argo
```
Comment (alexec) [inline manifests/install.yaml:200]: this looks wrong to me

Comment (dseynaev) [reply manifests/install.yaml:200]: caused by https://github.com/argoproj/argo-workflows/blob/7dc6515ce1ef76475ac7bd2a7a3c3cdbe795a13c/manifests/cluster-install/kustomization.yaml

Comment (dseynaev) [reply manifests/install.yaml:200]: introduced here: https://github.com/argoproj/argo-workflows/pull/5907

**Review by alexec — COMMENTED** [review]
I've fixed the `make codegen` on master. Can you re-run `make codegen` on your branch - should fix it.

**Review by alexec — APPROVED** [review]

**Pull request merged**